### PR TITLE
remove parameter and optimize cdn data source.

### PR DESCRIPTION
**Pull request by crab21**
remove parameter `offset` and optimize cdn data source.
***
$ make testacc TESTARGS=" -run=TestAccTencentCloudCdnDomains"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudCdnDomains -timeout 120m
?   	github.com/terraform-providers/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudCdnDomains
--- PASS: TestAccTencentCloudCdnDomains (129.58s)
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud	129.608s
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

### CODE REVIEW

File: tencentcloud/service_tencentcloud_cdn.go [hunk 1]
```diff
@@ -106,42 +112,37 @@ func (me *CdnService) DescribeDomainsConfigByFilters(ctx context.Context,
...
-	request.Offset = &offset
-
-	ratelimit.Check(request.GetAction())
-	response, err := me.client.UseCdnClient().DescribeDomainsConfig(request)
-	if err != nil {
-		if sdkErr, ok := err.(*errors.TencentCloudSDKError); ok {
-			if sdkErr.Code == CDN_HOST_NOT_FOUND {
-				return
+	for {
+		request.Limit = &limit
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_cdn.go:116]: request.Limit offset赋值放在for循环外边就可以了

File: tencentcloud/service_tencentcloud_cdn.go [hunk 2]
```diff
@@ -106,42 +112,37 @@ func (me *CdnService) DescribeDomainsConfigByFilters(ctx context.Context,
...
+		request.Offset = &offset
+
+		err = resource.Retry(readRetryTimeout, func() *resource.RetryError {
+			ratelimit.Check(request.GetAction())
+			response, err = me.client.UseCdnClient().DescribeDomainsConfig(request)
+
+			if err != nil {
+				if sdkErr, ok := err.(*errors.TencentCloudSDKError); ok {
+					if sdkErr.Code == CDN_HOST_NOT_FOUND {
+						return resource.NonRetryableError(err)
```
Comment (Himer) [inline tencentcloud/service_tencentcloud_cdn.go:126]: sdkErr.Code == CDN_HOST_NOT_FOUND  不要返回error 直接return nil就可以了

File: tencentcloud/service_tencentcloud_cdn.go (hunk 1 above)
Comment (crab21) [reply tencentcloud/service_tencentcloud_cdn.go]: > request.Limit offset赋值放在for循环外边就可以了

fixed

File: tencentcloud/service_tencentcloud_cdn.go (hunk 2 above)
Comment (crab21) [reply tencentcloud/service_tencentcloud_cdn.go]: > sdkErr.Code == CDN_HOST_NOT_FOUND 不要返回error 直接return nil就可以了

fixed

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+## 1.40.4 (Unreleased)
+
+BUG FIXES:
+
+* Data Source: `tencentcloud_cdn_domains` remove parameter `offset`.
```
Comment (gailwang) [inline CHANGELOG.md:5]: 废弃应该是有标志的写法？叫Deprecated

**Pull request merged**